components/project-list/project-list.jsx

```diff
@@ -11,47 +8,55 @@ export default React.createClass({
   getInitialState() {
     return {
       dataLoaded: false,
-      apiPageIndex: 1,
       entries: [],
       displayBatchIndex: 1
     };
   },
+  getDefaultProps() {
+    return {
+      params: {
+        featured: null,
+        issue: null,
+        ids: null
+      }
+    };
+  },
   componentDidMount() {
     this.fetchData(this.props.params);
   },
   componentWillReceiveProps(nextProps) {
-    // <ProjectList> doesn't always get unmounted and remounted when navigating between pages.
-    // (e.g., When navigating between the /issue/issue-name pages <Issue><ProjectList></Issue> do not
-    //  get re-mounted since the same components are being rendered.)
-    // It is treated as passing new props to theses components (<Issue><ProjectList></Issue>)
-    // By calling this.fetchData() in the componentWillReceiveProps method here
-    // we ensure data gets fetched and displayed accordingly.
+    // Reset state
+    this.resetState();
 
-    if ( this.props.params.issue ) {
-      // reset this.state.entries
-      this.setState({
-        entries: []
-      });
+    // <ProjectList> doesn't always get unmounted and remounted when navigating between pages.
+    // (e.g., When navigating between the /issue/issue-name pages <Issue><ProjectList></Issue>
+    //        do not get re-mounted since the same components are being rendered.)
+    // It is treated as passing new props to <ProjectList {...newProps} />
+    let newIssueEntered = (nextProps.params.issue !== this.props.params.issue);
 
-      // fetch data based on the new params props
+    if ( newIssueEntered ) {
```

Comment:
from `routes.jsx`

```jsx
<Route path="issues">
  <IndexRoute component={Issues} />
  <Route path=":issue" component={Issue} />
</Route>
```

And my inline comments in `project-list.jsx`
```jsx
// <ProjectList> doesn't always get unmounted and remounted when navigating between pages.
// (e.g., When navigating between the /issue/issue-name pages <Issue><ProjectList></Issue>
//        do not get re-mounted since the same components are being rendered.)
// It is treated as passing new props to <ProjectList {...newProps} />
```

And `issue` prop is being used in the next line after the one we are commenting here on
```jsx
if ( newIssueEntered ) {
  // Fetch data based on the new params props to ensure data gets fetched and displayed accordingly.
  this.fetchData(nextProps.params);
}
```